autogen/coding/jupyter/docker_jupyter_server.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+
+from pathlib import Path
+import sys
+from time import sleep
+from types import TracebackType
+import uuid
+from typing import Dict, Optional, Union
+import docker
+import secrets
+import io
+import atexit
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
+
+
+from .jupyter_client import JupyterClient
+from .base import JupyterConnectable, JupyterConnectionInfo
+
+
+KERNEL_DOCKERFILE = """FROM quay.io/jupyter/docker-stacks-foundation
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+USER ${NB_UID}
+RUN mamba install --yes jupyter_kernel_gateway ipykernel && \
+    mamba clean --all -f -y && \
+    fix-permissions "${CONDA_DIR}" && \
+    fix-permissions "/home/${NB_USER}"
+
+ENV TOKEN="UNSET"
+CMD python -m jupyter kernelgateway --KernelGatewayApp.ip=0.0.0.0 \
+    --KernelGatewayApp.port=8888 \
+    --KernelGatewayApp.auth_token="${TOKEN}" \
+    --JupyterApp.answer_yes=true \
+    --JupyterWebsocketPersonality.list_kernels=true
+
+EXPOSE 8888
```

Comment:
Should the port be configurable?

Reply:
It's just the port used internally in the dockerfile. The external facing port is configurable in the class constructor